api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,38 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  try {
+    const accessToken = req.get('X-Access-Token');
+
+    if (!accessToken) {
+      throw new ClientError(401, 'authentication required');
+    }
+    const payload = jwt.verify(accessToken, process.env.TOKEN_SECRET);
+
+    req.user = payload;
+
+    next();
+
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
Be sure to pass the `err` to `next()` so that Express can continue and finish handling the request. Otherwise the client will hang and timeout while your server leaks socket connections.